stellargraph/random.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2020 Data61, CSIRO
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import random as rn
+import numpy.random as np_rn
+from collections import namedtuple
+
+
+RandomState = namedtuple("RandomState", "random, numpy")
+
+
+def _global_state():
+    return RandomState(rn, np_rn)
+
+
+def _seeded_state(s):
+    return RandomState(rn.Random(s), np_rn.RandomState(s))
+
+
+_rs = _global_state()
+
+
+def random_state(seed):
```

Comment:
Is this function meant to be used by an external user?

Reply:
Hmm I guess not, are you suggesting it should be defined with a leading underscore in the name?

Reply:
The distinction I've made currently is that `_global_state` and `_seeded_state` are only used within this file, whereas the other two are used outside (but not necessary "user"-facing)

Reply:
Yeah, but I guess just making sure it's excluded from the docs is probably fine too. Maybe there could be an `__all__ = ["set_seed"]` at the top, potentially with a comment about excluding `random_state`?